estransport/estransport.go

```diff
@@ -141,23 +196,26 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 	}
 
 	for i := 1; i <= c.maxRetries; i++ {
+		// fmt.Printf("Attempt %d\n", i)
 		var (
-			nodeURL     *url.URL
+			conn        *Connection
 			shouldRetry bool
 		)
 
-		// Get URL from the Selector
+		// Get connection from the pool
 		//
-		nodeURL, err = c.getURL()
+		conn, err = c.getConnection()
```

Comment:
This makes me wonder if this is really a connection pool, or a clever URL selector. Using the stdlib Transport, we have the actual connections and connection pool handled in another layer. Plus: there is no limit of active connections, neither here, nor in the default stdlib pool.

Reply:
Correct, this is basically a "smart URL selector". I went with the naming to keep it in line with the other official clients, since it's "on the fence" between a selector and connection pool.